pkg/images/enricher/enricher_impl.go

```diff
@@ -238,6 +238,22 @@ func (e *enricherImpl) EnrichImage(ctx context.Context, enrichContext Enrichment
 
 	errorList := errorhelpers.NewErrorList("image enrichment")
 
+	//verify that there is an integration for  scanner specified in enrichContext
+	if enrichContext.ScannerTypeHint != "" {
+		scanners := e.integrations.ScannerSet()
+		found := false
+		for _, scanner := range scanners.GetAll() {
+			if scanner.GetScanner().Type() != enrichContext.ScannerTypeHint {
+				found = true
+				break
+			}
+		}
+		if !found {
+			errorList.AddError(errors.New("no image scanners are integrated"))
+			return EnrichmentResult{ImageUpdated: false, ScanResult: ScanNotDone}, errorList.ToError()
+		}
+	}
+
```

Comment:
Please move this logic into the `enrichWithScan` method, so that if no scan is performed it does not break enrichment (which it does current state)

Also, as written this only validates that a scanner exists of the specified type, it does not ensure the scanner with that type is used. In the `enrichWithScan` method will also need to ensure that the scanner of this type is used after verifying that a scanner with this type exists.